### PR TITLE
Feat/apply nextjs version checker

### PR DESCRIPTION
**Pull request by ChernoBen**
- static class that checks nextjs version

### CODE REVIEW

File: src/utils/version-checker/node-checker/node-checker.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Versions } from "./versions/versions";
+import { InvalidVersion } from "./errors/errors";
+
+export class NodeChecker{
+
+    public static check():boolean{
```
Comment (olivmath) [inline src/utils/version-checker/node-checker/node-checker.ts:6]: linter here

**Review by olivmath — CHANGES_REQUESTED** [review]

**Review by MagnunAVFAzion — DISMISSED** [review]
LGTM 🤘

File: src/utils/version-checker/version-checker.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export class VersionChecker{
+
+    public static node_version():boolean {
+        return NodeChecker.check();
+    }
+
+    public static nextjs_version(target_dir:string):boolean {
+
+        if(this.project_type(target_dir) != "nextjs"){
+            return true;
```
[the remark this answers is not in the document]
Comment (ChernoBen) [reply src/utils/version-checker/version-checker.ts:16]: This method just needs to ensure that a given project is a azion nextjs project. So if its not, the verification ends at this stage.

File: src/main.ts [hunk 3]
```diff
@@ -13,6 +15,12 @@ const { version } = require('../package.json');
 
 program.version(version);
 
+try{
```
[the remark this answers is not in the document]
Comment (enicio) [reply src/main.ts:18]: displayError just logs a console.log of the error, maybe that's why it doesn't stop the program. I think try catch wouldn't need to involve the whole program.

**Review by olivmath — DISMISSED** [review]

**Review by MagnunAVFAzion — APPROVED** [review]

**Review by enicio — APPROVED** [review]

**Pull request merged**